jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpDateTime.java

```diff
@@ -0,0 +1,240 @@
+                if (month == -1 && token.length() == 3)
+                {
+                    Integer m = MONTH_CACHE.getBest(token);
+                    if (m != null)
+                    {
+                        month = m;
+                        continue;
+                    }
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.4
+                if (year == -1)
+                {
+                    if (token.length() <= 2)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    else if (token.length() == 4)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    continue;
+                }
+            }
+        }
+        catch (Throwable x)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Ignore: Unable to parse Date/Time", x);
+        }
+
+        // RFC 6265 - Section 5.1.1 - Step 3
+        if ((year > 70) && (year <= 99))
+            year += 1900;
+        // RFC 6265 - Section 5.1.1 - Step 4
+        if ((year >= 0) && (year <= 69))
+            year += 2000;
+
+        // RFC 6265 - Section 5.1.1 - Step 5
+        if (day == -1)
+            throw new IllegalArgumentException("Missing [day]: " + datetime);
+        if (month == -1)
+            throw new IllegalArgumentException("Missing [month]: " + datetime);
+        if (year == -1)
+            throw new IllegalArgumentException("Missing [year]: " + datetime);
+        if (year < 1601)
+            throw new IllegalArgumentException("Too far in past [year]: " + datetime);
+        if (hour == -1)
+            throw new IllegalArgumentException("Missing [time]: " + datetime);
+        if (day < 1 || day > 31)
+            throw new IllegalArgumentException("Invalid [day]: " + datetime);
+        if (month < 1 || month > 12)
+            throw new IllegalArgumentException("Invalid [month]: " + datetime);
+        if (hour > 23)
+            throw new IllegalArgumentException("Invalid [hour]: " + datetime);
+        if (minute > 59)
+            throw new IllegalArgumentException("Invalid [minute]: " + datetime);
+        if (second > 59)
+            throw new IllegalArgumentException("Invalid [second]: " + datetime);
+
+        // RFC 6265 - Section 5.1.1 - Step 6
+        ZonedDateTime dateTime = ZonedDateTime.of(year,
+            month, day, hour, minute, second, 0, GMT);
+
+        // RFC 6265 - Section 5.1.1 - Step 7
+        return dateTime;
+    }
+
+    /**
+     * Formats provided Date/Time to a String following preferred RFC 1123 syntax from
+     * both HTTP and Cookie specs.
+     *
+     * @param datetime the date/time to format
+     * @return the String representation of the date/time
+     */
+    public static String format(TemporalAccessor datetime)
+    {
+        return DateTimeFormatter.RFC_1123_DATE_TIME
+            .withZone(GMT)
```

Comment:
UTC

Reply:
Actually this one should be "GMT", as we are meant to generate as GMT, but interpret as UTC.  I guess theoretically we should format with UTC and replace the "UTC" with "GMT", but that would only be significant during leap seconds.  So I'm fine with GMT here.  But above we should use UTC, as per the specification.

Reply:
left this as GMT.

Reply:
Reverted this.
Now it's UTC as well.
Also, there are not format() test cases.